cobra/io/sbml.py

```diff
@@ -1057,14 +1052,20 @@ def _model_to_sbml(cobra_model, f_replace=None, units=True):
             if f_replace and F_GENE_REV in f_replace:
                 gpr = gpr.replace('(', '( ')
                 gpr = gpr.replace(')', ' )')
+                gpr = gpr.replace('  ', ' ')
                 tokens = gpr.split(' ')
+
                 for k in range(len(tokens)):
                     if tokens[k] not in [' ', 'and', 'or', '(', ')']:
                         tokens[k] = f_replace[F_GENE_REV](tokens[k])
                 gpr_new = " ".join(tokens)
+            else:
+                gpr_new = gpr
 
             gpa = r_fbc.createGeneProductAssociation()  # noqa: E501 type: libsbml.GeneProductAssociation
-            gpa.setAssociation(gpr_new)
+            # uses ids for GP and does not create GPs
+            _check(gpa.setAssociation(gpr_new, True, False),
```

Comment:
What do the new arguments change here?

Reply:
Added better documentation:
```
# uses ids to identify GeneProducts (True),
# does not create GeneProducts (False)
```
The first True decides if `label` or `id` should be used to identify the GeneProduct. To be more specific and avoid problems in case of gene labels I specified to always use the ids. The second flag allows to create the GeneProducts from a GPR if these are not existing in the model. We add all GeneProducts before, so no need to add these.
This could fix some bugs if people have labels on genes, so added the flags while fixing things.